extensions/sql-database-projects/src/models/project.ts

```diff
@@ -120,62 +120,95 @@ export class Project implements ISqlProject {
 
 		// check if this is a new msbuild sdk style project
 		this._isMsbuildSdkStyleProject = this.CheckForMsbuildSdkStyleProject();
+
 		// get projectGUID
-		this._projectGuid = this.projFileXmlDoc!.documentElement.getElementsByTagName(constants.ProjectGuid)[0].childNodes[0].nodeValue!;
+		try {
+			this._projectGuid = this.projFileXmlDoc!.documentElement.getElementsByTagName(constants.ProjectGuid)[0].childNodes[0].nodeValue!;
+		} catch {
+			void window.showErrorMessage(constants.errorReadingProject(constants.ProjectGuid, this.projectFilePath));
+		}
 
 		// find all folders and files to include
 		for (let ig = 0; ig < this.projFileXmlDoc!.documentElement.getElementsByTagName(constants.ItemGroup).length; ig++) {
 			const itemGroup = this.projFileXmlDoc!.documentElement.getElementsByTagName(constants.ItemGroup)[ig];
 
-			const buildElements = itemGroup.getElementsByTagName(constants.Build);
-			for (let b = 0; b < buildElements.length; b++) {
-				this._files.push(this.createFileProjectEntry(buildElements[b].getAttribute(constants.Include)!, EntryType.File, buildElements[b].getAttribute(constants.Type)!));
+			try {
+				const buildElements = itemGroup.getElementsByTagName(constants.Build);
+				for (let b = 0; b < buildElements.length; b++) {
+					this._files.push(this.createFileProjectEntry(buildElements[b].getAttribute(constants.Include)!, EntryType.File, buildElements[b].getAttribute(constants.Type)!));
+				}
+			} catch {
```

Comment:
The smaller try catches are so that the other parts of the project can still be loaded even if one section fails. For example, even if a sqlproj doesn't have a project guid, the rest of the project will still be loaded and shown in the project viewlet. The current behavior is that if the project guid is missing, the rest of the project doesn't load.

Reply:
Good point!